Framework/Algorithms/src/Qxy.cpp

```diff
@@ -392,8 +391,12 @@ Qxy::setUpOutputWorkspace(API::MatrixWorkspace_const_sptr inputWorkspace) {
   // Fill the X vectors in the output workspace
   for (int i = 0; i < bins - 1; ++i) {
     outputWorkspace->setBinEdges(i, axis);
-    outputWorkspace->mutableY(i) = std::numeric_limits<double>::quiet_NaN();
-    outputWorkspace->mutableE(i) = std::numeric_limits<double>::quiet_NaN();
+    for (int j = 0; j < bins - j; ++j) {
+      outputWorkspace->mutableY(i)[j] =
```

Comment:
Just like the old `dataY` methods `mutableY` is slightly expensive (due to the copy-on-write mechanism), so it should be avoided to call it in a loop for each bin (for each histogram). That is, try to store a reference to it before the loop over `j`.